app/src/main/java/com/example/concordia_campus_guide/Models/Building.java

```diff
@@ -2,35 +2,75 @@
 
 import android.text.TextUtils;
 
+import androidx.annotation.NonNull;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.Ignore;
+import androidx.room.TypeConverters;
+
+import com.example.concordia_campus_guide.Database.Converters.StringListConverter;
 import com.google.android.gms.maps.model.GroundOverlayOptions;
 
 import org.json.JSONArray;
 import org.json.JSONObject;
 
 import java.util.List;
 
+@Entity (tableName = "buildings",
+        primaryKeys = {"building_code"}
+        )
 public class Building extends Place {
     /**
      * These following attributes NEED to match the fields in the JSON file. Do not change them.
      */
-    //Attributes needed to create the ground overlays [ui]
-    private String[] availableFloors;
-    private Float width;
-    private Float height;
-    private Float bearing;
+
+    //Needs to be modified to list of floors instead.
+    @ColumnInfo(name = "availableFloors")
+    @TypeConverters(StringListConverter.class)
+    private List<String> availableFloors;
+
+    @ColumnInfo(name = "width")
+    private float width;
+
+    @ColumnInfo(name = "height")
+    private float height;
+
+    @ColumnInfo(name = "bearing")
+    private float bearing;
+
+    //Not sure what to do here yet
```

Comment:
Didn't we say we'll change the structure of the coordinates variable so that it won't be a List of a list of a list of double?

Reply:
easier said than done :p